Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,276 @@
+    {
+      std::cerr << "Error: Compression not in acceptable range for the format" << std::endl;
+      return 1;
+    }
+  }
+
+  // Open the blob reader
+  std::unique_ptr<DiscIO::BlobReader> blob_reader =
+      scrub ? DiscIO::ScrubbedBlob::Create(input_file_path) :
+              DiscIO::CreateBlobReader(input_file_path);
+  if (!blob_reader)
+  {
+    std::cerr << "Error: Unable to process disc image. If --scrub is enabled, try again without it."
+              << std::endl;
+    return 1;
+  }
+
+  // Perform the conversion
+  const auto NOOP_STATUS_CALLBACK = [](const std::string& text, float percent) { return true; };
+
+  std::future<bool> success;
+
+  switch (format)
+  {
+  case DiscIO::BlobType::PLAIN:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToPlain(blob_reader.get(), input_file_path, output_file_path,
+                                               NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::GCZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good =
+          DiscIO::ConvertToGCZ(blob_reader.get(), input_file_path, output_file_path,
+                               volume->GetVolumeType() == DiscIO::Platform::WiiDisc ? 1 : 0,
+                               block_size_o.value(), NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::WIA:
+  case DiscIO::BlobType::RVZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToWIAOrRVZ(
+          blob_reader.get(), input_file_path, output_file_path, format == DiscIO::BlobType::RVZ,
+          compression_o.value(), compression_level_o.value(), block_size_o.value(),
+          NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  default:
+    ASSERT(false);
+    break;
+  }
+
+  if (!success.get())
+  {
+    std::cerr << "Error: Conversion failed" << std::endl;
+    return 1;
+  }
+
+  std::cout << "Success" << std::endl;
+  return 0;
+}
+
+bool ConvertCommand::CheckBlockSize(const int block_size, const u64 file_size,
+                                    const DiscIO::BlobType format)
+{
+  if (format == DiscIO::BlobType::GCZ)
+  {
+    // Refer to ConvertDialog.cpp
+    if (!(file_size % block_size == 0 &&
+          file_size % (block_size * ConvertCommand::GCZ_BLOCKS_PER_BUFFER) != 0))
+    {
+      std::cerr << "Error: Invalid block size for GCZ. File size must be an integer multiple of "
+                   "the block size "
+                   "and must not be an integer multiple of the block size multiplied by 32.";
+      return false;
+    }
+  }
+
+  for (int good_block_size = ConvertCommand::MIN_BLOCK_SIZE;
+       good_block_size <= ConvertCommand::MAX_BLOCK_SIZE; good_block_size *= 2)
+  {
+    if (block_size == good_block_size)
+      return true;
+  }
+
+  return false;
```

Comment:
Perhaps consider having just a warning in this case as well, but I care less about this case than the previous case.

Reply:
Ah, but you would still have to filter out sizes smaller than 32 KiB for RVZ. And if you allow sizes larger than 2 MiB for WIA/RVZ, you should ideally also implement checking if the size is an integer multiple of 2 MiB rather than a power of two... So maybe you should just leave this part as it is.

Reply:
One thing I did notice just now is that you don't seem to be filtering out sizes smaller than 2 MiB for WIA. Am I missing where it is, or is it not implemented?

Reply:
Hi, thanks for the tips. Should be much better now. Check IsDiscImageBlockSizeCompatible() in DiscUtils